script.js

```diff
@@ -1,4 +1,184 @@
+var playerHand = []; //all cards player will hold
+var computerHand = []; // all cards computer will hold
+var gameMode = 'deal2Cards';
+
+// create deck
+var makeDeck = function () {
+  var deck = [];
+  var suitIndex = 0;
+  var suits = ["hearts", "diamonds", "clubs", "spades"];
+  while (suitIndex < suits.length) {
+    var currentSuit = suits[suitIndex];
+    //console.log("current suit: " + currentSuit);
+    var rankCounter = 1;
+    while (rankCounter <= 13) {
+      var cardName = rankCounter;
+      if (cardName == 1) {
+        cardName = "ace";
+      } else if (cardName == 11) {
+        cardName = "jack";
+      } else if (cardName == 12) {
+        cardName = "queen";
+      } else if (cardName == 13) {
+        cardName = "king";
+      }
+      var card = {
+        name: cardName,
+        suit: currentSuit,
+        rank: rankCounter,
+      };
+      //console.log("rank: " + rankCounter);
+      deck.push(card);
+      rankCounter += 1;
+    }
+    suitIndex += 1;
+  }
+  return deck;
+};
+
+// shuffle deck 
+var getRandomIndex = function (max) {
+  return Math.floor(Math.random() * max);
+};
+
+// shuffle the deck and save it in a new variable shuffledDeck
+// to communicate that we have shuffled the deck.
+var shuffleCards = function (cardDeck) {
+  var currentIndex = 0;
+  while (currentIndex < cardDeck.length) {
+    var randomIndex = getRandomIndex(cardDeck.length);
+    var randomCard = cardDeck[randomIndex];
+    var currentCard = cardDeck[currentIndex];
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard;
+    currentIndex = currentIndex + 1;
+  }
+  return cardDeck;
+};
+
+var deck = makeDeck();
+var shuffledDeck = shuffleCards(deck);
+
+
+//function to take a hand and come up with points
+//add rank
+// count aces
+// write the score
+var calculate = function(hand){
+  var currentScore = 0;
+  var cardIndex = 0;
+  var handLength = hand.length;
+  var gotAce = 0;
+  while (cardIndex < handLength){
+    var currentCard = hand[cardIndex]
+    var cardScore = currentCard.rank
+    if (cardScore == 1){
+      gotAce += 1;
+    }
+    currentScore += Math.min(10,cardScore)
+    cardIndex += 1
+  }
+  if (gotAce > 0 & currentScore <= 11){
+    currentScore += 10;
+  }
+  return currentScore
+}
```

Comment:
good use of a helper function and naming. A slightly more description name would be: calculateHand